### PR TITLE
test: Loader Component 100% Test Coverage and Fix Uncovered Lines

### LINKED ISSUE #1026: Code Coverage: Create tests for src/components/Loader/Loader.tsx
We will need unittests done for all methods, classes and functions found in this file.

Any widgets, components, modals referenced in this file must also have unittests done

### PR Acceptance Criteria

- When complete this file must show at least **100%** coverage when merged into the code base. 
- [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin?search=&displayType=list)
- The PR will show a report for the code coverage for the file you have added. You can use that as a guide
- Please coordinate **issue assignment** and **PR reviews** with the contributors listed in this issue https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by akhilender-bongirwar**
**What kind of change does this PR introduce?**

- Improved the test coverage for the Loader component, addressing the previously uncovered lines and ensuring that all tests pass successfully.
- Added a test to ensure that the component renders correctly with a custom 'sm' size. I verified that both the spinner-wrapper and spinner elements are present and that the correct class is applied.
- Added a test to ensure that the component renders correctly with a 'lg' size. I verified that the spinner element has the appropriate class.
- Added a test to ensure that the component renders correctly with an 'xl' size. I verified that the spinner element has the correct class.

**Issue Number:**

Fixes  #1026 

**Did you add tests for your changes?**
 Yes

**Snapshots/Videos:**
**Before**
prevTest
**After**
testsLoader100

**If relevant, did you update the documentation?**
No

**Summary**
With these new tests, I now have 100% test coverage, and there are no more uncovered lines. The Loader component is thoroughly tested for different size scenarios, and all tests pass successfully.

**Does this PR introduce a breaking change?**
 No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by akhilender-bongirwar** [comment]
@Ayush0Chaudhary, could you please review this PR .

**Comment by Ayush0Chaudhary** [comment]
@akhilender-bongirwar please write the tests, and bring them to acceptable level

**Review by noman2002 — APPROVED** [review]
Looks good to me. Expect the acceptable code coverage is failing because its 95% and its not possible to reach there in this PR. We can merge this.

**Comment by noman2002** [comment]
@rishav-jha-mech @beingnoble03 Please review once.

File: src/components/Loader/Loader.test.tsx [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
-        <Loader />
-      </BrowserRouter>
-    );
+    render(<Loader />);
 
-    expect(screen.getByTestId('spinner-wrapper')).toBeInTheDocument();
+    const spinnerWrapper = screen.getByTestId('spinner-wrapper');
+    expect(spinnerWrapper).toBeInTheDocument();
     expect(screen.getByTestId('spinner')).toBeInTheDocument();
+    expect(spinnerWrapper).toHaveClass('_spinner_wrapper_1vy2z_1');
```
Comment (beingnoble03) [inline src/components/Loader/Loader.test.tsx:12]: You should not include these classes which are generated on build time. Please remove this check.
```suggestion
```

File: src/components/Loader/Loader.test.tsx [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
+  });
+
+  test('Component should render on custom sizes', () => {
+    render(<Loader size="sm" />);
+
+    expect(screen.getByTestId('spinner')).toHaveClass(
+      '_spinnerSm_1vy2z_21 spinner-border text-primary'
```
Comment (beingnoble03) [inline src/components/Loader/Loader.test.tsx:19]: Same remove `_spinnerSm_1vy2z_21` from check.

File: src/components/Loader/Loader.test.tsx [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
+    );
+  });
+
+  test('Component should render with large size', () => {
+    render(<Loader size="lg" />);
+
+    expect(screen.getByTestId('spinner')).toHaveClass(
+      '_spinnerLg_1vy2z_15 spinner-border text-primary'
```
Comment (beingnoble03) [inline src/components/Loader/Loader.test.tsx:27]: Same remove `_spinnerLg_1vy2z_15` from check.

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
Some changes.

**Comment by akhilender-bongirwar** [comment]
@beingnoble03, This is my first time writing tests, and I initially believed that checking for classNames also would enhance the test coverage in the codecov report. But validating classes assigned during runtime is inappropriate. Thank you for bringing this to my attention.  I appreciate your feedback  : )

**Review by beingnoble03 — APPROVED** [review]
LGTM.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**